openeo_processes_dask/process_implementations/comparison.py

```diff
@@ -0,0 +1,140 @@
+from typing import Callable, Optional
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from numpy.typing import ArrayLike
+from xarray.core.duck_array_ops import notnull
+
+__all__ = [
+    "is_infinite",
+    "is_valid",
+    "is_nodata",
+    "is_nan",
+    "eq",
+    "neq",
+    "gt",
+    "gte",
+    "lt",
+    "lte",
+    "between",
+]
+
+
+def is_infinite(x: ArrayLike):
+    if x is None:
+        return False
+    if (
+        type(x) in [str, list, dict]
+        or type(x) in [np.ndarray, da.core.Array]
+        and x.dtype.kind.lower() in ["u", "s", "o"]
+    ):
+        return False
+    return np.isinf(x)
+
+
+def is_valid(x: ArrayLike):
+    finite = np.logical_not(is_infinite(x))
+    return np.logical_and(notnull(x), finite)
+
+
+def is_nodata(x: ArrayLike):
+    return np.logical_not(is_valid(x))
+
+
+def is_nan(x: ArrayLike):
+    return is_nodata(x)
+
+
+def eq(
+    x: ArrayLike,
+    y: ArrayLike,
+    delta: Optional[float] = None,
+    case_sensitive: Optional[bool] = True,
+):
+    if not type(x) in [np.ndarray, da.core.Array] and not type(y) in [
+        np.ndarray,
+        da.core.Array,
+    ]:
+        if is_nodata(x) or is_nodata(y):
+            return np.nan
+    if x is False or y is False:
```

Comment:
I think this line isn't correct, eq(False, False) should be True!

Reply:
Hm, I think this came from the example of eq(x = 0, y = false) => false in https://processes.openeo.org/#eq. 
I think, I had assumed that if one incoming value is false, the output should be as well. But I probably interpreted it wrong, what do you think?

Reply:
> I had assumed that if one incoming value is false the output should be as well

I think you're thinking of the short circuiting that is required for the boolean processes (e.g. `and` being immediately False if the first value is False), but I don't think it's appropriate here!

The example `eq(x = 0, y = false) => false` is thus because the integer 0 is not equal to the boolean False